Content.Server/Lathe/LatheSystem.cs

```diff
@@ -54,217 +57,172 @@ public override void Update(float frameTime)
-
-            // TODO: You can probably split this part off of lathe component too
-            if (!TryComp<LatheComponent>(uid, out var lathe))
-                return;
-
-            // We need the prototype to get the color
-            _prototypeManager.TryIndex(lastMat, out MaterialPrototype? matProto);
+                Recipes = component.StaticRecipes
+            };
+            RaiseLocalEvent(component.Owner, ev, true);
+            return ev.Recipes;
+        }
 
-            EntityManager.QueueDeleteEntity(args.Used);
+        public List<string> GetAllBaseRecipes(LatheComponent component)
+        {
+            return component.DynamicRecipes == null
+                ? component.StaticRecipes
+                : component.StaticRecipes.Union(component.DynamicRecipes).ToList();
+        }
 
-            EnsureComp<LatheInsertingComponent>(uid).TimeRemaining = lathe.InsertionTime;
+        public bool TryAddToQueue(EntityUid uid, LatheRecipePrototype recipe, LatheComponent? component = null)
+        {
+            if (!Resolve(uid, ref component))
+                return false;
 
-            _popupSystem.PopupEntity(Loc.GetString("machine-insert-item", ("machine", uid),
-                ("item", args.Used)), uid, Filter.Entities(args.User));
+            if (!CanProduce(uid, recipe, 1, component))
+                return false;
 
-            if (matProto != null)
+            foreach (var (mat, amount) in recipe.RequiredMaterials)
             {
-                UpdateInsertingAppearance(uid, true, matProto.Color);
+                _materialStorage.TryChangeMaterialAmount(uid, mat, -amount);
             }
-            UpdateInsertingAppearance(uid, true);
-        }
+            component.Queue.Add(recipe);
 
-
-        private void OnPowerChanged(EntityUid uid, LatheComponent component, PowerChangedEvent args)
-        {
-            //if the power state changes, try to produce.
-            //aka, if you went from unpowered --> powered, resume lathe queue.
-            TryStartProducing(uid, component: component);
+            return true;
         }
 
-        /// <summary>
-        /// This handles the checks to start producing an item, and
-        /// starts up the sound and visuals
-        /// </summary>
-        private bool TryStartProducing(EntityUid uid, LatheProducingComponent? prodComp = null, LatheComponent? component = null)
+        public bool TryStartProducing(EntityUid uid, LatheComponent? component = null)
         {
-            if (!Resolve(uid, ref component) || component.Queue.Count == 0)
+            if (!Resolve(uid, ref component))
                 return false;
-
-            if (!this.IsPowered(uid, EntityManager))
+            if (HasComp<LatheProducingComponent>(uid) || !component.Queue.Any())
                 return false;
 
-            var recipeId = component.Queue[0];
+            var recipe = component.Queue.First();
+            component.Queue.RemoveAt(0);
```

Comment:
Probably better to just make it a stack as that's what it functions as IMO.

Reply:
As far as I know, it cannot be a stack because the recipes are taken off the front of the queue and added to the end. From what i can see, stacks only support popping off the top.